include/image-commands.mk

```diff
@@ -210,8 +210,8 @@ define Build/check-size
 	@imagesize="$$(stat -c%s $@)"; \
 	limitsize="$$(($(subst k,* 1024,$(subst m, * 1024k,$(if $(1),$(1),$(IMAGE_SIZE))))))"; \
 	[ $$limitsize -ge $$imagesize ] || { \
-		$(call ERROR_MESSAGE,    WARNING: Image file $@ is too big: $$imagesize > $$limitsize); \
-		rm -f $@; \
```

Comment:
why the rm is removed?

Also I would keep the call and just add the exit 1; if we really want to make this thing critical...

What I would love to know is where check-size is used as an intermediate command instead of checking it on the final image. Aka I want to know where the File not found error is present.

Reply:
Some images will check-size on the kernel section to ensure it isn't too big for soft-splitting partitions later, where soft-splitting is used (uboot has mtd partition "firmware" of whole size, but the mtd under linux splits that into kernel/rootfs during early boot, with a fixed size for the kernel section.)  These will crash with premature end of data on the kernel during lzma-loader expansion, if the kernel goes outside the partition.